### ISSUE TITLE
Hibernate mappings of spring sessionFactory are ignored

### ISSUE THREAD

**Issue by malaguna** [liquibase/liquibase-hibernate#61]
I have a project using Spring and Hibernate. Hibernate persistence is defined into spring bean sessionFactory. I am using *xml hibernate mappings* instead annotations or JPA.

If I configure *referenceUrl* property as `hibernate:spring:applicationCtx.xml?bean=sessionFactory`, liquibase-hibernate find the applicaction context, detects dialect, but it ignores mappings:

```
[main] INFO liquibase-hibernate - Reading hibernate configuration hibernate:spring:applicationCtx.xml?bean=sessionFactory
Aug 16, 2014 11:22:42 AM org.hibernate.annotations.common.reflection.java.JavaReflectionManager <clinit>
INFO: HCANN000001: Hibernate Commons Annotations {4.0.5.Final}
Aug 16, 2014 11:22:42 AM org.hibernate.Version logVersion
INFO: HHH000412: Hibernate Core {4.3.6.Final}
Aug 16, 2014 11:22:42 AM org.hibernate.cfg.Environment <clinit>
INFO: HHH000206: hibernate.properties not found
Aug 16, 2014 11:22:42 AM org.hibernate.cfg.Environment buildBytecodeProvider
INFO: HHH000021: Bytecode provider name : javassist
Aug 16, 2014 11:22:42 AM org.springframework.beans.factory.xml.XmlBeanDefinitionReader loadBeanDefinitions
INFO: Loading XML bean definitions from class path resource [applicationCtx.xml]
Aug 16, 2014 11:22:42 AM org.hibernate.dialect.Dialect <init>
INFO: HHH000400: Using dialect: org.hibernate.dialect.HSQLDialect
[main] INFO liquibase-hibernate - Using dialect org.hibernate.dialect.HSQLDialect
```
However, if I create a fake hibernate config file with the same mappings than sessionFactory of application context, it works perfectly with *referenceUrl* as `hibernate:classic:liquibase/fake-hibernate.cfg.xml`:

```
[main] INFO liquibase-hibernate - Reading hibernate configuration hibernate:classic:liquibase/fake-hibernate.cfg.xml
Aug 16, 2014 11:00:34 AM org.hibernate.annotations.common.reflection.java.JavaReflectionManager <clinit>
INFO: HCANN000001: Hibernate Commons Annotations {4.0.5.Final}
Aug 16, 2014 11:00:34 AM org.hibernate.Version logVersion
INFO: HHH000412: Hibernate Core {4.3.6.Final}
Aug 16, 2014 11:00:34 AM org.hibernate.cfg.Environment <clinit>
INFO: HHH000206: hibernate.properties not found
Aug 16, 2014 11:00:34 AM org.hibernate.cfg.Environment buildBytecodeProvider
INFO: HHH000021: Bytecode provider name : javassist
Aug 16, 2014 11:00:34 AM org.hibernate.cfg.Configuration configure
INFO: HHH000043: Configuring from resource: liquibase/fake-hibernate.cfg.xml
Aug 16, 2014 11:00:34 AM org.hibernate.cfg.Configuration getConfigurationInputStream
INFO: HHH000040: Configuration resource: liquibase/fake-hibernate.cfg.xml
Aug 16, 2014 11:00:35 AM org.hibernate.cfg.Configuration addResource
INFO: HHH000221: Reading mappings from resource: es/myapp/infraestructura/model/usrmgt/Log.hbm.xml
Aug 16, 2014 11:00:35 AM org.hibernate.cfg.Configuration addResource
INFO: HHH000221: Reading mappings from resource: es/myapp/infraestructura/model/usrmgt/User.hbm.xml
Aug 16, 2014 11:00:35 AM org.hibernate.cfg.Configuration doConfigure
```
I am using a custom liquibase-maven-pluging 3.3.0-SNAPSHOT with msvticket's "CORE-1987" patch and liquibase-hibernate 3.4

**Comment by malaguna** [comment]
Recently I tried once again and now I could see where the bug is. HibernateSpringDatabase imports `LocalSessionFactory` from `org.springframework.orm.hibernate3` package and I am using `LocalSessionFactory` from `org.springframework.orm.hibernate4`.

I forked **liquibase-hibernate4** and change it and now it is working like a charm. ¿could you see if this change is applicable? It seems to be compatible with `sessionFactories` that uses `org.springframework.orm.hibernate3.LocalSessionFactory`, but I didn't test it very well.

**Comment by nvoxland** [comment]
Fixed with pull request #83
